mountpoint-s3/tests/fuse_tests/fork_test.rs

```diff
@@ -484,6 +486,160 @@ fn mount_with_sse(
+        .spawn()
+        .expect("unable to spawn child");
+
+    let exit_status = wait_for_exit(child);
+
+    // verify mount status and mount entry
+    assert!(exit_status.success());
+    assert!(mount_exists("mountpoint-s3", mount_point.path().to_str().unwrap()));
+
+    test_read_files(&bucket, &prefix, &region, &mount_point.to_path_buf());
+
+    unmount(mount_point.path());
+
+    Ok(())
+}
+
+#[test]
+fn mount_with_assumed_role_in_other_region() -> Result<(), Box<dyn std::error::Error>> {
+    let (bucket, prefix) = get_test_bucket_and_prefix("mount_with_assumed_role_in_other_region");
+    let subsession_role = get_subsession_iam_role();
+    let region = get_test_region();
+    let other_region = get_random_non_test_region();
+    let mount_point = assert_fs::TempDir::new()?;
+    let profile_name = "fork_test";
+
+    let mut config_file = NamedTempFile::new()?;
+    writeln!(config_file, "[profile {}]", profile_name).unwrap();
+    writeln!(config_file, "role_arn = {}", subsession_role).unwrap();
+    writeln!(config_file, "source_profile = default").unwrap();
+    writeln!(config_file, "region = {}", other_region).unwrap();
+
+    let mut cmd = Command::cargo_bin("mount-s3")?;
+    let child = cmd
+        .arg(&bucket)
+        .arg(mount_point.path())
+        .arg(format!("--prefix={prefix}"))
+        .arg("--auto-unmount")
+        .env("AWS_CONFIG_FILE", config_file.path())
+        .env("AWS_PROFILE", profile_name)
+        .spawn()
+        .expect("unable to spawn child");
+
+    let exit_status = wait_for_exit(child);
+
+    // verify mount status and mount entry
+    assert!(exit_status.success());
+    assert!(mount_exists("mountpoint-s3", mount_point.path().to_str().unwrap()));
+
+    test_read_files(&bucket, &prefix, &region, &mount_point.to_path_buf());
+
+    unmount(mount_point.path());
+
+    Ok(())
+}
+
+// Test profile auth without setting a region so it will fallback to global STS endpoint
+#[test]
+fn mount_with_assumed_role_no_region() -> Result<(), Box<dyn std::error::Error>> {
+    let (bucket, prefix) = get_test_bucket_and_prefix("mount_with_assumed_role_no_region");
+    let subsession_role = get_subsession_iam_role();
+    let region = get_test_region();
+    let mount_point = assert_fs::TempDir::new()?;
+    let profile_name = "fork_test";
+
+    let mut config_file = NamedTempFile::new()?;
+    writeln!(config_file, "[profile {}]", profile_name).unwrap();
+    writeln!(config_file, "role_arn = {}", subsession_role).unwrap();
+    writeln!(config_file, "source_profile = default").unwrap();
+
+    let mut cmd = Command::cargo_bin("mount-s3")?;
+    let child = cmd
+        .arg(&bucket)
+        .arg(mount_point.path())
+        .arg(format!("--prefix={prefix}"))
+        .arg("--auto-unmount")
+        .env("AWS_CONFIG_FILE", config_file.path())
+        .env("AWS_PROFILE", profile_name)
+        .env_remove("AWS_REGION")
+        .env_remove("AWS_DEFAULT_REGION")
```

Comment:
Can you also add a env var to disable IMDS, so this test would still work if CRT add it to their region chain?

```suggestion
        .env("AWS_CONFIG_FILE", config_file.path())
        .env("AWS_PROFILE", profile_name)
        .env("AWS_EC2_METADATA_DISABLED", "true")
        .env_remove("AWS_REGION")
        .env_remove("AWS_DEFAULT_REGION")
```